RecoMuon/L2MuonSeedGenerator/src/Phase2L2MuonSeedCreator.cc

```diff
@@ -0,0 +1,842 @@
+
+        // Propagate matched segments to the seed and try to extrapolate in unmatched chambers
+        std::vector<int> matchedStations;
+        matchedStations.reserve(4);
+        for (auto& [detId, matchingPair] : matchesInBarrel) {
+          // Add matched segments to the seed
+          LogDebug(metname) << "Adding matched DT segment in station " << detId.station() << " to the seed";
+          container.push_back(dtSegments[matchingPair.first]);
+          matchedStations.push_back(detId.station());
+        }
+        for (int station = 1; station != 5; ++station) {
+          if (std::find(matchedStations.begin(), matchedStations.end(), station) == matchedStations.end()) {
+            // Try to extrapolate from stations with a match to the ones without
+            LogDebug(metname) << "No matching DT segment found in station " << station;
+            auto extrapolatedMatch = extrapolateToNearbyStation(station, matchesInBarrel, dtSegments);
+            if (extrapolatedMatch.first != -1) {
+              LogDebug(metname) << "Adding extrapolated DT segment " << extrapolatedMatch.first << " with quality "
+                                << extrapolatedMatch.second << " found in station " << station << " to the seed";
+              container.push_back(dtSegments[extrapolatedMatch.first]);
+            }
+          }
+        }
+      } else if (!bestInDt) {
+        // Found a matching segment in CSC -> propagate to ME2
+        LogDebug(metname) << "Found matching segment(s) in CSCs, propagating L1TkMu info to ME2 to seed";
+        // ME2
+        propagateToId = eta > 0 ? CSCDetId(1, 2, 0, 0, 0) : CSCDetId(2, 2, 0, 0, 0);
+        detLayer = service_->detLayerGeometry()->idToLayer(propagateToId);
+        radius = std::abs(detLayer->position().z() / cos(theta));
+
+        // Fill seed with matched segment(s)
+        for (auto& [detId, matchingPair] : matchesInEndcap) {
+          LogDebug(metname) << "Adding matched CSC segment in station " << detId.station() << " to the seed";
+          container.push_back(cscSegments[matchingPair.first]);
+        }
+      }
+      vec.setMag(radius);
+      // Get Global point and direction
+      GlobalPoint pos(vec.x(), vec.y(), vec.z());
+      GlobalVector mom(pt * cos(phi), pt * sin(phi), pt * cos(theta) / sin(theta));
+
+      GlobalTrajectoryParameters param(pos, mom, charge, &*magneticField_);
+
+      AlgebraicSymMatrix55 mat;
+
+      mat[0][0] = bestInDt ? (0.25 / pt) * (0.25 / pt) : (0.4 / pt) * (0.4 / pt);  // sigma^2(charge/abs_momentum)
+      mat[1][1] = 0.05 * 0.05;                                                     // sigma^2(lambda)
+      mat[2][2] = 0.2 * 0.2;                                                       // sigma^2(phi)
+      mat[3][3] = 20. * 20.;                                                       // sigma^2(x_transverse)
+      mat[4][4] = 20. * 20.;                                                       // sigma^2(y_transverse)
```

Comment:
where are all these numbers coming from?

Reply:
The propagation to the various muon stations have remained basically unchanged from the [current L2 MuonSeedGenerator](https://github.com/cms-sw/cmssw/blob/87f3f24501aefe75d4e0be6f3f624e890b8a621d/RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGenerator.cc#L241) (seeded by L1Tk Muons)

Reply:
The aim is for this new module to replace the current "legacy" seeding strategy. However, it's currently under a process modifier as a soft introduction while some slight inefficiencies are ironed out and its performance is assessed thoroughly.  In any case, in the end, there should be a single Standalone Muon seed from L1Tk Muons generator